rainflame/geopolygonize
# Log errors per-process and fix exception

Followup to PR #43 to fix issue #33.

This PR does the following:
1. Log errors per-process instead of in one file to prevent the possibility of races.
2. The logs revealed that the `TopologyException` being used was wrong, so I used a `shapely` exception that encompasses this class of exception.

**Test**

1. Run `./tests/run_e2e_test.sh central_oregon`.
2. The CLI outputs the following, indicating it encountered the `TopologyException`:
```
Encountered GEOSException: TopologyException: side location conflict at -43954.42796256701 47360.496962901969. This can occur if the input geometry is invalid.
```
3. Observe that there are no logs in the log directory, indicating that there are no unanticipated errors.

After stitching, we get:
```
geopolygonize encountered error: TopologyException: side location conflict at 28048.042625668284 -24642.797154745087. This can occur if the input geometry is invalid.
```
I'll fix this in a followup PR.
